sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs

```diff
@@ -532,7 +532,11 @@ private void EnsureState()
                     Resource = Constants.Sas.Resource.BlobVersion;
                 }
             }
-            Version = SasQueryParameters.DefaultSasVersion;
+
+            if (string.IsNullOrEmpty(Version))
```

Comment:
This is 1 of 2 logic changes I made in customer-facing code.  This if check is present in Data Lake, File Shares, and Queues, but was not present in blob for some reason.

Reply:
This was conscious change here https://github.com/Azure/azure-sdk-for-net/pull/15972 . Which means we forgot other places earlier. @amnguye please take a look.